tinygrad/lazy.py

```diff
@@ -255,7 +255,7 @@ def splitted_shape(dim_aft_div): return self.shape[:dim_to_split] + (self.shape[
   # *** movement ops ***
 
   def _movement_op(self, st: ShapeTracker, op: MovementOps, arg: Union[Tuple[sint, ...], Tuple[Tuple[sint, sint], ...]]) -> LazyBuffer:
-    if SHUFFLE_MOVEMENT_OPS and self.optype == BinaryOps and not self.realized and (op in {MovementOps.SHRINK, MovementOps.STRIDE, MovementOps.PERMUTE} or (op == MovementOps.RESHAPE and self.op.op in UnaryOps)) and not self.children:
+    if SHUFFLE_MOVEMENT_OPS and not self.realized and self.optype == BinaryOps and len(self.children) <= 1 and (op is not MovementOps.PAD or all(x.op not in UNSAFE_PAD_OPS for x in self.op.get_lazyops())) and not all(x.realized for x in self.op.buffers):
```

Comment:
Not sure you want `len(self.children) <= 1` here, `not self.children` is safer.

Reply:
restored